### PR TITLE
CY-1681 Use ssl client certs for snapshot restore

### PR DESCRIPTION
**Pull request by geokala**

### CODE REVIEW

File: cfy_manager/components/composer/composer.py [hunk 1]
```diff
@@ -45,8 +45,10 @@
 NODEJS_DIR = join('/opt', 'nodejs')
 LOG_DIR = join(BASE_LOG_DIR, COMPOSER)
 
-DB_CLIENT_KEY_PATH = join(CONF_DIR, 'db_client.key')
-DB_CLIENT_CERT_PATH = join(CONF_DIR, 'db_client.crt')
+# These are all the same key as the other db keys, but postgres is very strict
+# about permissions (no group or other permissions allowed)
+DB_CLIENT_KEY_PATH = join('/etc/cloudify/ssl/composer_db.key')
+DB_CLIENT_CERT_PATH = join('/etc/cloudify/ssl/composer_db.crt')
```
Comment (tehasdf) [inline cfy_manager/components/composer/composer.py]: join a single literal path? presumably you meant some constant?

File: cfy_manager/components/mgmtworker/mgmtworker.py [hunk 2]
```diff
@@ -54,21 +54,31 @@ class MgmtWorker(BaseComponent):
...
-            get_local_source_path(premium_source_url)
-        except FileError:
-            logger.info(
-                'premium package not found in manager resources package')
-            logger.notice('premium will not be installed.')
-        else:
-            logger.notice('Installing Cloudify Premium...')
-            cluster = Cluster(skip_installation=False)
-            cluster.install()
+        scripts = {
```
Comment (tehasdf) [inline cfy_manager/components/mgmtworker/mgmtworker.py]: tbh a list of pairs would work just as well

File: cfy_manager/components/mgmtworker/mgmtworker.py [hunk 3]
```diff
@@ -144,7 +155,20 @@ def _configure(self):
 
     def install(self):
         logger.notice('Installing Management Worker...')
-        self._install()
+        source_url = config[MGMTWORKER][SOURCES]['mgmtworker_source_url']
```
Comment (tehasdf) [inline cfy_manager/components/mgmtworker/mgmtworker.py]: this seems to just be moving it, yes, no change?

File: cfy_manager/components/mgmtworker/scripts/allow-snapshot-ssl-client-cert-access [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#! /usr/bin/env bash
+chown cfyuser. /etc/cloudify/ssl/stage_db.*
+chown cfyuser. /etc/cloudify/ssl/composer_db.*
```
Comment (tehasdf) [inline cfy_manager/components/mgmtworker/scripts/allow-snapshot-ssl-client-cert-access]: this is just two items, yes? the cert and the key?
i'm somewhat afraid of globs in sudo, maybe let's just write out the two files?

File: cfy_manager/components/mgmtworker/mgmtworker.py (hunk 3 above)
Comment (geokala) [reply cfy_manager/components/mgmtworker/mgmtworker.py]: Yup.

File: cfy_manager/components/mgmtworker/scripts/allow-snapshot-ssl-client-cert-access (hunk 4 above)
Comment (geokala) [reply cfy_manager/components/mgmtworker/scripts/allow-snapshot-ssl-client-cert-access]: This isn't in sudo, it's in the script hat is being run already through sudo- but yes, point taken.

File: cfy_manager/components/mgmtworker/mgmtworker.py (hunk 2 above)
Comment (geokala) [reply cfy_manager/components/mgmtworker/mgmtworker.py]: Changed.

File: cfy_manager/components/composer/composer.py (hunk 1 above)
Comment (geokala) [reply cfy_manager/components/composer/composer.py]: Good catch, changed to just make them constants.

**Pull request merged**